### PR TITLE
ben-nanonote: drop

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

This does not evaluate for at least 2 years. The hardware is 12 years
old. MIPS is a dead horse. Let's focus on the important things.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
Motivation? Seems this is even used in a normal derivation in nixpkgs.

**Comment by Mic92** [comment]
Seems like the motivation I added to the commit  was committed the first time. Fixed this now.

**Comment by sternenseemann** [comment]
You'd need to drop `xburst-tools` as well then.

**Comment by viric** [comment]
xbursttools builds fine and updated one year ago. What is wrong with the ben nanonote?

**Comment by r-burns** [comment]
Various issues with uclibc, apparently. Hopefully https://github.com/NixOS/nixpkgs/pull/149953 will get it back in working order.

I haven't been paying close attention to eval failures on release-cross, I should take a look at these. I definitely agree with removing vaporware jobs but I bet a lot of these are easy to fix :)

**Comment by Mic92** [comment]
> You'd need to drop `xburst-tools` as well then.

We can replace by the other mips target we have. It only needs the compiler not the libc, that does not evaluate.

**Comment by Mic92** [comment]
> Various issues with uclibc, apparently. Hopefully #149953 will get it back in working order.
> 
> I haven't been paying close attention to eval failures on release-cross, I should take a look at these. I definitely agree with removing vaporware jobs but I bet a lot of these are easy to fix :)

I tried to fix it for a bit... but it was too much work for outdated hardware.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10982,7 +10982,7 @@ with pkgs;
   xbursttools = callPackage ../tools/misc/xburst-tools {
     # It needs a cross compiler for mipsel to build the firmware it will
     # load into the Ben Nanonote
-    gccCross = pkgsCross.ben-nanonote.buildPackages.gccCrossStageStatic;
+    gccCross = pkgsCross.fuloongminipc.buildPackages.gccCrossStageStatic;
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:10985]: This hopefully fixes eval.

**Comment by Mic92** [comment]
ben-nanonote gets to live another day: https://github.com/NixOS/nixpkgs/pull/149953
Also I rather would name those targets closer to what architecture they stand for rather than specific hardware...

**Pull request closed without merging**